core/web-assets/src/main/assets/js/apps/onms-endpoints/grafana/grafana-modal.html

```diff
@@ -0,0 +1,95 @@
+<style type="text/css">
+  .rotate {
+    -webkit-animation: rotation 2s infinite linear;
+  }
+
+  @-webkit-keyframes rotation {
+    from {
+      -webkit-transform: rotate(0deg);
+    }
+    to {
+      -webkit-transform: rotate(359deg);
+    }
+  }
+</style>
+
+
+<div id="endpointModal">
+  <div class="modal-header">
+    <h5 class="modal-title">{{endpoint.id === undefined ? "Add" : "Edit"}} Grafana Endpoint</h5>
+  </div>
+  <div class="modal-body">
+    <form name="endpointForm" novalidate>
+      <div class="form-group">
+        <div ng-show="error.entity" class="form-text text-danger">{{error.entity}}</div>
+      </div>
+      <div class="form-group form-row">
+        <label class="col-form-label" for="endpoint.uid">Grafana ID</label>
+        <input class="form-control" type="text" id="endpoint.uid" name="uid" placeholder="Unique ID" ng-model="endpoint.uid"
```

Comment:
When editing this item in the modal, the changes are also reflected immediately in the table that's in the background. These shouldn't change until the values are valid and saved.

Reply:
Done